### PR TITLE
add rating, modal, text & grid component, updated button 

### PR DESCRIPTION
**Pull request by govindappaarun**
- added rating component
 - added modal component
 -  add typography related component
 - refactored button component
 - add documentation

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -16,11 +16,22 @@
...
     <div class="container">
-   <section id="alert">
+     <section>
+       <h2 class="h2">Rating </h2>
+      <pre>
+        <code class="language-html">
+          <div class="stars" style="--rating:3.5;"></div>
+        </code>
+      </pre> 
+      <div class="stars" style="--rating:3.5; background-color: rebeccapurple;"></div>
```
Comment (zoef-yt) [inline index.html:30]: Could you please explain why you used the inline styling when you have also used class?

File: src/common/box.css [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+.my-2 {
+  margin: 2rem 0;
+}
+
+.h-100 {
+  height: 100px;
+}
+
+.h-200 {
+  height: 200px;
```
Comment (zoef-yt) [inline src/common/box.css]: rem can be used here.

File: src/common/flex.css [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+.flex-ai-center {
+  align-items: center;
+}
+
+.flex-center-column {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  flex-direction: column;
+}
```
Comment (zoef-yt) [inline src/common/flex.css:57]: Adding spacing between line 56 and 57

File: src/common/flex.css [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+.flex-gap-0 {
+  gap: 0.5rem;
+}
```
Comment (zoef-yt) [inline src/common/flex.css:60]: Adding spacing between line 59 and 60

File: src/common/helper.css [hunk 5]
```diff
@@ -1,86 +1,4 @@
...
-  gap: 5rem;
-}
-
-.flex-grow-1 {
-  flex-grow: 1;
-}
-
 .shadow-xs {
   box-shadow: 0px 2px 1px -1px rgba(0, 0, 0, 0.2),
     0px 1px 1px 0px rgba(0, 0, 0, 0.14), 0px 1px 3px 0px rgba(0, 0, 0, 0.12) !important;
```
Comment (zoef-yt) [inline src/common/helper.css]: Use var for colours.

File: src/components/button/index.css [hunk 6]
```diff
@@ -58,6 +58,18 @@ input[type="reset"] {
 }
 .btn-error:hover {
   background-color: var(--error-dark);
+  transform: translateY(-3px);
+}
+.btn-warning {
+  background-color: var(--warning-main);
+  border-color: var(--warning-main);
+  color: white;
```
Comment (zoef-yt) [inline src/components/button/index.css]: Use var for colours.

File: src/components/button/index.css [hunk 7]
```diff
@@ -154,3 +183,15 @@ input[type="reset"] {
   box-shadow: 2px 1px 8px 0px rgba(0, 0, 0, 0.75);
   color: white;
```
Comment (zoef-yt) [inline src/components/button/index.css]: Use var for colours.

File: src/components/modal/index.css [hunk 8]
```diff
@@ -0,0 +1,41 @@
+/* https: //css-tricks.com/considerations-styling-modal/; */
+.modal {
+  position: fixed;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+  height: 400px;
+  width: 600px;
```
Comment (zoef-yt) [inline src/components/modal/index.css]: use rem instead of px

File: src/components/modal/index.css [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+  max-width: 100%;
+  z-index: 1010;
+
+  background: white;
+  box-shadow: 0 0 60px 10px rgba(0, 0, 0, 0.9);
```
Comment (zoef-yt) [inline src/components/modal/index.css]: Use var for colours.

File: src/components/modal/index.css [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+.modal-overlay {
+  z-index: 1000;
+  position: fixed;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  /* background-color: var(--grey-ultra); */
```
Comment (zoef-yt) [inline src/components/modal/index.css]: remove the commented code.

File: src/components/modal/index.css [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+  background: rgba(0, 0, 0, 0.6);
```
Comment (zoef-yt) [inline src/components/modal/index.css]: Use var for colours.

**Review by zoef-yt — CHANGES_REQUESTED** [review]

File: src/common/box.css (hunk 2 above)
Comment (govindappaarun) [reply src/common/box.css]: inline styling was to give a custom value and you get a option of specifying an explicit value

File: src/common/helper.css (hunk 5 above)
Comment (ShekharDhangar) [inline src/common/helper.css]: try to remove the important tag

**Review by ShekharDhangar — APPROVED** [review]
Rest is good

**Review by devsurajrai — APPROVED** [review]
Did not find any breaking changes. Although you can try to use semantic html wherever possible. Everything else looks good. You have used !important once in your code try if you can fix the issue with specificity.

**Review by VLeads — APPROVED** [review]
Some suggestions are made by peers also. Rest all code looks good !

**Pull request merged**